datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -517,13 +517,12 @@ def run(self):
                 from ..utils.agent.debug import enter_pdb
 
                 enter_pdb(self.check, line=self.init_config['set_breakpoint'], args=(instance,))
-            elif 'profile_memory' in self.init_config:
-                from ..utils.agent.memory import TRACE_LOCK, profile_memory
+            elif 'profile_memory' in self.init_config or datadog_agent.tracemalloc_enabled():
```

Comment:
Wondering if this is the right way to handle this... shouldn't this be an `and` condition? I think I know why you're doing this (for instance when running the check command?), but if `tracemalloc` isn't enabled on the agent side, you'll likely have multiple runners and the profiling will be a mess?

Reply:
Yes for the check command, in which case only one check will ever being running, no?

Reply:
OK, given that this is injected by the check command, we should be good to go!